kbase-extension/static/kbase/js/widgets/narrative_core/upload/importErrors.js

```diff
@@ -0,0 +1,246 @@
+         *
+         * i.e. from:
+         * [{
+         *   type: 'cannot_find_file',
+         *   file: 'some_file.csv',
+         * }]
+         * to:
+         * {
+         *   some_file.csv: ['not found']
+         * }
+         * or, more complicated:
+         * [{
+         *   type: 'incorrect_column_count',
+         *   file: 'file1.xls',
+         *   tab: 'tab_1',
+         *   message: 'wrong column count, missing <some column>'
+         * }, {
+         *   type: 'multiple_specifications_for_data_type',
+         *   file_1: 'file1.xls',
+         *   tab_1: 'tab_1',
+         *   file_2: 'file2.csv',
+         *   tab_2: null,
+         *   message: 'duplicate specification found'
+         * }, {
+         *   type: 'unexpected_error',
+         *   message: 'an unexpected error occurred'
+         * }]
+         * to:
+         * {
+         *   'file1.xls tab "tab_1"': [
+         *     'wrong column count, missing <some column>',
+         *     'duplicate specification found in file2.csv'
+         *   ],
+         *   'file2.csv': [
+         *     'duplicate specification found in file1.xls tab "tab_1"'
+         *   ],
+         * }
+         */
+        _processErrors() {
+            // a little helper to handle file error formatting
+            const addFileError = (error) => {
+                let fileKey = error.file;
+                if (error.tab) {
+                    fileKey += ` tab "${error.tab}"`;
+                }
+                if (!this.fileErrors[fileKey]) {
+                    this.fileErrors[fileKey] = [];
+                }
+                this.fileErrors[fileKey].push(error.message);
+            };
+
+            this.errors.forEach((error) => {
+                switch (error.type) {
+                    case BULK_SPEC_ERRORS.CANNOT_PARSE:
+                    case BULK_SPEC_ERRORS.INCORRECT_COLUMN_COUNT:
+                        addFileError(error);
+                        break;
+                    case BULK_SPEC_ERRORS.NOT_FOUND:
+                        addFileError(Object.assign({ message: 'File not found' }, error));
+                        break;
+                    case BULK_SPEC_ERRORS.MULTIPLE_SPECS:
+                        addFileError({
+                            file: error.file_1,
+                            tab: error.tab_1,
+                            message: error.message,
+                        });
+                        addFileError({
+                            file: error.file_2,
+                            tab: error.tab_2,
+                            message: error.message,
+                        });
+                        break;
+                    case BULK_SPEC_ERRORS.NO_FILES:
+                        this.noFileError = true;
+                        break;
+                    case BULK_SPEC_ERRORS.UNKNOWN:
+                        if (error.file) {
+                            addFileError(error);
```

Comment:
will this always have an `error.message` field?

Reply:
Not necessarily... I coulda sworn I put in a default message for that if not. Will fix.

Reply:
Oh, I misread that. Yes, it's expected to have a message, with type `unexpected_error` from the staging service. Still, that feels generic enough that it could use a generic message.

Reply:
Fixed with a default error.